### ISSUE TITLE
[DONATION] Kepler Light Curve Classification

### ISSUE THREAD

**Issue by benfulcher** [uea-machine-learning/tsml_repo#80]
**Please give a brief description of the data**

Light curves from NASA's _Kepler_ mission (3-month-long series, sampled every ~30 min: 4767 samples per time series), hand-labeled by experts in asteroseismology from The University of Sydney. A seven-class, class-imbalanced problem.

**Where can we get the data?**

Contact us directly for the time-series data (ben.fulcher@sydney.edu.au).

**Is there a publication related to the data**

Journal publication is [here](https://academic.oup.com/mnras/article/514/2/2793/6598817); preprint is [here](https://arxiv.org/abs/2205.03020). The paper shows the results of a feature-based approach using [hctsa](https://github.com/benfulcher/hctsa).

**Is your data multivariate or univariate?**

Univariate.

**Is there a default train/test split for the data**

No.

**Are there any missing values/padding?**

All series are an equal length. Missing values have already been interpolated.

**Is there a picture for the data we can put on the website?**

Yes: <img width="1051" alt="Screen Shot 2022-07-12 at 11 41 43" src="https://user-images.githubusercontent.com/4446070/178389871-33a8051c-d502-4407-8dd8-deda46f18a0e.png">

**How would you like the donation attributed on the website?**

Processed dataset donated by: Nicholas H. Barbara, Timothy R. Bedding, Ben D. Fulcher, Simon J. Murphy and Timothy Van Reeth. We thank the [Kepler](https://www.nasa.gov/mission_pages/kepler/overview/index.html) team for providing the [original and wonderful dataset](https://kasoc.phys.au.dk/).

**Can you provide a paragraph description, including the meaning of the class values, for the website?**

Light curves (brightness of an object sampled over time) from NASA's _Kepler_ mission, focusing on A and F-type stars with eﬀective temperatures in the range 6500–10,000 K. We label seven classes: 𝛿 Scuti stars, 𝛾 Doradus stars, RR Lyrae stars, rotational variables, contact eclipsing binaries, detached eclipsing binaries, and non-variable stars. Each light curve corresponds to 3 months of data, sampled every 29.45 min, resulting in 4767 samples per time series. Further information can be found in the original paper: Barbara et al. (2022) Classifying Kepler light curves for 12 000 A and F stars using supervised feature-based machine learning. Monthly Notices of the Royal Astronomical Society. (preprint available at: https://arxiv.org/abs/2205.03020). Note that this paper refers to the dataset provided here as the 'training set'.

**Comment by TonyBagnall** [comment]
superb, thanks Ben, I am finally trying to get on top of the repo and update the website, despite my woeful webdev skills. If you have any more, we would love them, even if its just a link to where we could get them. Particularly looking for more large problems and high dimensional multivariate problems. Also interested in any time series regression (not forecasting) problems.

**Comment by benfulcher** [comment]
No worries—and from a first-time contributor, thanks for maintaining this really important resource! :)
Ok I'll keep it in mind—we have started some work in the multivariate space and could contribute (or encourage others) to share their neuroscience data.
For this one, let me know the best format to share the processed dataset…

**Comment by ChangWeiTan** [comment]
This is great! I would be interested in more larger TSC problems as well as regression ones. Looking forward to these.

**Comment by TonyBagnall** [comment]
@ChangWeiTan @benfulcher I'm away at a conference next week, but I will have another pass at the archives the week after and start with this donation. I'm mostly focussing on getting multivariate up to 50, but we have some new large univariate datasets too. I also have a new project starting in October on EEG/MEG classification, so more neuroscience data most welcome.

**Comment by TonyBagnall** [comment]
@benfulcher so obviously got distracted and then went on holiday! We can sort this now though. The best format is a csv file where each row is an observation and the class label is at the end. We can then easily put it into arff and ts format.  I would like to provide a default split, is there anything that we should take into account (e.g. order of observations) or can we just do a random 50/50 split?

**Comment by benfulcher** [comment]
Hi there. No worries at all about the delay—thanks for getting back to it. The student has just set up the csv file (attached).

Class labels are integers as:

1 = Contact eclipsing binary
2 = Detached eclipsing binary
3 = Delta Scuti variable
4 = Gamma Doradus variable
5 = Non variable
6 = Rotational variable
7 = RR Lyrae variable

**Split**

Regarding the split, as some of the training classes have quite varied behavior (eg, the delta Scutis), we think a 70% (training)—30% (test) split would be the most appropriate, stratified (so that each class has 70% of its examples in the training set). Am I right that you're happy to do the split? Let me know if you'd prefer us to do this split on our end. [As an additional comment, it's quite a imbalanced, seven-class problem, so raw 'classification accuracy' is probably going to be a deceptive metric for performance].

Otherwise, let me know if there are any additional questions and we'll help you out :)

[kepler_feature_matrix_training.csv](https://github.com/uea-machine-learning/tsml_repo/files/9486327/kepler_feature_matrix_training.csv)

**Comment by TonyBagnall** [comment]
I am happy to do the split, so we use camel case naming, is KelperLightCurve ok as a name?

**Comment by TonyBagnall** [comment]
I am happy to do the split, but I think something went wrong in the processing, series are length 5!
![image](https://user-images.githubusercontent.com/9594042/188825086-6255d71e-e749-43e0-8e95-2360884f8296.png)

**Comment by benfulcher** [comment]
Oop that's embarrassing (these are 5 extracted features from each series)—apologies 🙈
`KeplerLightCurve` is a fine name.
See if this works to download the .csv (128MB), which should now be in the right format, with class labels at the very right of each row:
https://www.dropbox.com/s/13dxzkhsh1q9n5m/training_lightcurves_and_classes.csv?dl=0

**Comment by TonyBagnall** [comment]
yep, got it, will try sort it out next week.

**Comment by TonyBagnall** [comment]
data is up there now, I did a stratified resampling for 70/30. https://timeseriesclassification.com/description.php?Dataset=KeplerLightCurves could you take a look and if its ok, close this issue? We will include it in our experiments for a group of new data on the website

**Comment by benfulcher** [comment]
Hi Tony—great, thanks for it! A few pieces of feedback:

1. Can you update the pre-print link to be to the published paper? https://doi.org/10.1093/mnras/stac1515
2. The N-dash is rendering to a question mark, can perhaps change to a hyphen: "6500-10,000K"
3. Can edit the "Donated By" field to the set of authors that worked on this dataset: "Nicholas Barbara, Ben Fulcher, Timothy Bedding, Simon Murphy"

Hope it's not too much hassle. Thanks again! Hope to contribute more in future.

**Comment by TonyBagnall** [comment]
no problem, takes a minute, although the text parsing is always annoying. How is this? https://timeseriesclassification.com/description.php?Dataset=KeplerLightCurves

**Comment by benfulcher** [comment]
Thanks Tony! Great to have the data source link the main paper, but the second link in the description says it's a link to the preprint, so should stay as the arXiv link for anyone with access issues (sorry I should have been more specific). All good then! Cheers
